### ISSUE TITLE
[Bug] spatial_loss_maps in test_step computed without interior_mask, inconsistent with all other loss calls

### ISSUE THREAD

**Issue by RajdeepKushwaha5** [mllam/neural-lam#569]
**Description**

In `ar_model.py`, `test_step` computes `spatial_loss_maps` at L452-454 without passing `mask=self.interior_mask_bool`:

```python
spatial_loss = self.loss(
    prediction, target, pred_std, average_grid=False
)
```

Every other call to `self.loss` in the module passes `mask=self.interior_mask_bool`:

- `training_step` (L311)
- `validation_step` (L354)
- `test_step` main loss (L408)

The per-variable metrics (`mse`, `mae`) computed just above (L435-441) also pass the mask. This means the spatial loss maps include boundary grid nodes that are excluded everywhere else.

**Impact**

- The saved `mean_spatial_loss.pt` contains loss values at boundary nodes that the model never optimises for, which can mislead downstream error analysis.
- `plot_spatial_error` picks `vmin`/`vmax` from the full tensor, so boundary-node loss values can skew the colour scale.
- For LAM domains with boundary forcing the inconsistency is most pronounced; for global domains (no boundary) it has no practical effect.

**Minimal example**

```bash
python -m neural_lam.train_model --eval test --config_path <config> --load <ckpt>
# spatial_loss_maps saved to <logger_dir>/spatial_loss_maps/ include boundary values
```

**Proposed fix**

After computing spatial loss over the full grid (to preserve the `num_grid_nodes` shape needed by plotting), mask boundary nodes to NaN:

```python
spatial_loss = self.loss(
    prediction, target, pred_std, average_grid=False
)
spatial_loss[..., ~self.interior_mask_bool] = float("nan")
```

And use `torch.nanmean` instead of `torch.mean` when averaging the maps in `on_test_epoch_end`.
